### ISSUE TITLE
Issues with pantothenic acid and CO2/H2 consumptions

### ISSUE THREAD

**Issue by BizzoTL** [jotech/gapseq#113]
Hello,
I have been testing several media on a model of _Acetobacterium wodii_ and I have encountered some issues with some metabolites present in the growth medium. I am trying to optimize the growth rate while simultaneously producing methane by using both H2 and CO2. The medium also contains Acetate.
In particular, I had two separate issues encountered during the gapfill part:
1) If the medium with high H2 but no CO2 contains no Pantothenic acid methane is produced in high quantities (11,93) and the growth rate is 0,063. If the same medium contains Pantothenic acid(maxFlux=2) the growth rate slightly increases (0,068) but the production of methane drops to zero. The code I used:
`./gapseq fill -m Acetobacterium_woodii-draft.RDS -c Acetobacterium_woodii-rxnWeights.RDS -g Acetobacterium_woodii-rxnXgenes.RDS -n medium.csv -b 100`
However, if I add the argument  `-e highH2` the methane is produced in even higher quantities(16,2) both in the presence and absence of Pantothenic acid. As far as I am aware Panthothenic acid is not involved in methane production so I am a bit confused by the difference in production. Any ideas why this happens?

2) I had a similar issue when adding CO2 to the same medium(maxFlux=10). Using the normal gapfill code(as above) there is no difference in both growth rate and methane produced(0,063 and 11,93). When adding `-e highH2` the growth rate is diminished (0,007) and no methane is produced. My idea was to use both H2 and CO2 to increase the methane produced but the effect that I get is the opposite. I tried removing the other carbon sources from the medium(acetate, L-Tyrosine, formate) and there is no difference in the "highH2" case while there is some growth(0,015) and some methane(2,18) using the base command. Is there a way to "force" the use of both CO2 and H2 for both growth and methane production?

The gapseq version that I used is "1.1 2ec4c84". I also have attached the files written in the command.
[acetobacterium_model_and_medium.zip](https://github.com/jotech/gapseq/files/8370438/acetobacterium_model_and_medium.zip)

Sorry for the long post, any help is truly appreciated.

Best regards,
Edoardo

**Comment by Waschina** [comment]
Hi Edoardo,

I'll have a look at this case next week. What I usually do in such cases is to compare which reactions were gapfilled in each situation. Will let you know when I found something.

Best wishes
Silvio

**Comment by jotech** [comment]
Hi Edoardo,

I had a look at your example, it's an interesting case!
The problem is a bit related to the methanogenesis itself that operates at the thermodynamic limit of what is energetically feasible and therefore also makes gap filling more challenging! 

In your case, the gap filling step 1, which enables growth using all possible reactions, doesn't consider methanogenesis because, given your input, the fermentation of tyrosine and pathenoate is more favorable! The first gap filling step really tries to  permit the model to grow and depending on the quality of the draft network and the input medium can be prone to overshooting, especially when the desired growth is energetically challenging and the medium allows multiple growth conditions.

So basically the proportions of nutrients defined in medium file is quite crucial here! If you want to have a clear cut methanogen the easiest solution would be to remove or lower the available amount of tyrosine and pathenoate. This worked for me!

You might also have a look at the predefined media for methanogens:
`gapseq/dat/media/MM_anaerobic_CO2_H2.csv` and `gapseq/dat/media/MM_anaerobic_Acetate_H2.csv`

Let us know what you make out of it!

**Comment by jotech** [comment]
just as a side note: Acetobacterium woodii is not at all a methanogen! This could also contribute to the fact, that methanogenic genes are not infered and gap filling to methane production doesn't work well...
